plugins/woocommerce/includes/class-wc-brands.php

```diff
@@ -1093,6 +1095,80 @@ public function wc_brands_on_block_template_register( $template_id, $template_ar
 			}
 		}
 	}
+
+	/**
+	 * Hooks the product brand terms block into single product templates.
+	 * 
+	 * @param array $hooked_block_types The array of hooked block types.
+	 * @param int $relative_position The relative position of the hooked block.
+	 * @param string $anchor_block_type The type of anchor block.
+	 * @param WP_Block_Template $context The context of the block.
+	 * 
+	 * @return array The array of hooked block types.
+	 */
+	public function hook_product_brand_block( $hooked_block_types, $relative_position, $anchor_block_type, $context ) {
+
+		// Only add the block as the last child of the product meta block
+		if ( 'woocommerce/product-meta' === $anchor_block_type &&
+			 'last_child' === $relative_position &&
+			 $context instanceof WP_Block_Template &&
+			 'single-product' === $context->slug ) {
+
+				remove_action( 'woocommerce_product_meta_end', array( $this, 'show_brand' ) );
+
+				// Check if the template already has a product brand block
+				if ( ! $this->template_already_has_brand_block( $context ) ) {
+					// Simply add the core/post-terms block type
+					$hooked_block_types[] = 'core/post-terms';
+				}
+		}
+		
+		return $hooked_block_types;
+	}
+
+	/**
+	 * Check if the template already contains a product brand block.
+	 *
+	 * @param WP_Block_Template $template The template object.
+	 * 
+	 * @return boolean True if template contains a brand block.
+	 */
+	private function template_already_has_brand_block( $template ) {
+		if ( ! $template || empty( $template->content ) ) {
+			return false;
+		}
+
+		return strpos( $template->content, '<!-- wp:post-terms {"term":"product_brand"' ) !== false;
+	}
+
+	/**
+	 * Configures the attributes for the hooked product brand terms block.
+	 * 	
+	 * @param array $parsed_hooked_block The parsed hooked block.
+	 * @param string $hooked_block_type The type of hooked block.
+	 * @param int $relative_position The relative position of the hooked block.
+	 * @param array $parsed_anchor_block The parsed anchor block.
+	 * @param WP_Block_Template $context The context of the block.
+	 * 
+	 * @return array The parsed hooked block.
+	 */
+	public function configure_product_brand_block( $parsed_hooked_block, $hooked_block_type, $relative_position, $parsed_anchor_block, $context ) {
+		if ( is_null( $parsed_hooked_block ) ) {
+			return $parsed_hooked_block;
+		}
+
+		if ( 'core/post-terms' === $hooked_block_type &&
+			 'last_child' === $relative_position &&
+			 'woocommerce/product-meta' === $parsed_anchor_block['blockName'] ) {
+
+			$parsed_hooked_block['attrs'] = array(
+				'term'	 => 'product_brand',
+				'prefix' => __( 'Brands: ', 'woocommerce' ),
+			);
+		}
```

Comment:
Not sure what are the best practices with block hooks, but wondering if we should make sure `attrs` is empty before setting them to make sure we don't override an existing block? I'm not sure if that's even a problem, though, considering we use the `hooked_block_core/` hook. :thinking: 

cc @tjcafferkey in case you have a preference.

Reply:
Thanks @Aljullu. I think it doesn't hurt to add an extra check: [901ed79](https://github.com/woocommerce/woocommerce/pull/55936/commits/901ed798a49a37279dcdd85dbd4bed6299174c23)

Reply:
>Not sure what are the best practices with block hooks, but wondering if we should make sure attrs is empty before setting them to make sure we don't override an existing block?

This will never be an existing block. This filter only gets run for blocks which the API is hooking into the template. Once the block is updated or moved its no longer a hooked block. Adding the check is fine though and shows intent either way.